test/teiserver/microblog/microblog_test.exs

```diff
@@ -293,4 +293,142 @@ defmodule Teiserver.MicroblogTest do
+    end
+
+    test "update_poll_response/2 with valid data updates the poll_response" do
+      poll_response = poll_response_fixture()
+      update_attrs = %{}
+
+      assert {:ok, %PollResponse{} = _poll_response} =
+               Microblog.update_poll_response(poll_response, update_attrs)
+    end
+
+    test "update_poll_response/2 with invalid data returns error changeset" do
+      poll_response = poll_response_fixture()
+
+      assert {:error, %Ecto.Changeset{}} =
+               Microblog.update_poll_response(poll_response, @invalid_attrs)
+
+      assert poll_response ==
+               Microblog.get_poll_response(poll_response.user_id, poll_response.post_id)
+    end
+
+    test "delete_poll_response/1 deletes the poll_response" do
+      poll_response = poll_response_fixture()
+      assert {:ok, %PollResponse{}} = Microblog.delete_poll_response(poll_response)
+
+      assert Microblog.get_poll_response(poll_response.user_id, poll_response.post_id) == nil
+    end
+
+    test "change_poll_response/1 returns a poll_response changeset" do
+      poll_response = poll_response_fixture()
+      assert %Ecto.Changeset{} = Microblog.change_poll_response(poll_response)
+    end
+  end
+
+  describe "uploads" do
+    alias Teiserver.Microblog.Upload
+
+    import Teiserver.{AccountFixtures, MicroblogFixtures}
+
+    @invalid_attrs %{
+      filename: nil,
+      type: nil,
+      file_size: nil
+    }
+
+    test "list_uploads/0 returns all uploads" do
+      upload = upload_fixture()
+      assert Microblog.list_uploads() == [upload]
+    end
+
+    test "get_upload!/1 returns the upload with given id" do
+      upload = upload_fixture()
+      assert Microblog.get_upload!(upload.id) == upload
+    end
+
+    test "create_upload/1 with valid data creates a upload" do
+      user = user_fixture()
+
+      valid_attrs = %{
+        filename: "some contents",
+        type: "some title",
+        file_size: 42,
+        uploader_id: user.id
+      }
+
+      assert {:ok, %Upload{} = upload} = Microblog.create_upload(valid_attrs)
+      assert upload.filename == "some contents"
+      assert upload.type == "some title"
+      assert upload.file_size == 42
+    end
+
+    test "create_upload/1 with invalid data returns error changeset" do
+      assert {:error, %Ecto.Changeset{}} = Microblog.create_upload(@invalid_attrs)
+    end
+
+    test "update_upload/2 with valid data updates the upload" do
+      upload = upload_fixture()
+      user = user_fixture()
+
+      update_attrs = %{
+        filename: "some updated contents",
+        type: "some updated title",
+        file_size: 43,
+        uploader_id: user.id
+      }
+
+      assert {:ok, %Upload{} = upload} = Microblog.update_upload(upload, update_attrs)
+      assert upload.filename == "some updated contents"
+      assert upload.type == "some updated title"
+      assert upload.file_size == 43
```

Comment:
I don't think the last 3 asserts are of any value. Because `Repo.insert()` more or less returns the struct passed in, except for fields with `read_after_writes` or specifying `returning: xxx`.
If you want to test that the stuff in the db is indeed correct, you should query the record from the db, and perform asserts on that thing instead (or use the `:returning` feature)

Reply:
Fair